### PR TITLE
Status Reporting with k8s Patch

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Use kubectl patch rather than read then write pattern to update status on custom resources, which should improve gloo client rate limiting to the k8s apiserver. Also updates the default values for QPS/burst to the k8s apiserver to be 50/100 respectively, rather than 5/10 as they were before.

Depends on https://github.com/solo-io/solo-kit/pull/517

## Additional Cleanup

Removes some gloo `Settings` and `Configmap` permissions that were not needed:
- Configmap permissions were needed as of https://github.com/solo-io/gloo/pull/1274 but should have been removed in https://github.com/solo-io/gloo/pull/3850.
- Further the settings permissions should have been removed in https://github.com/solo-io/gloo/pull/1983.

## Upgrading

Users will have to apply the new CRDs so `status` and `status.statuses` are set by default on our CRs, or else status updates to new resources will fail. Once applied, no action is required from users as the CR gets / patches apply the default values before we attempt status patching. This is noted in the changelog for users, as well as in the error message logged by gloo for users who missed this guidance in the changelog here. If users forget to apply new CRDs, dataplane traffic and control plane updates will still work as normal; just status updates will fail on new CRs until the new CRDs have been applied.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick ssl config error? https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=0f06af5f-f328-48e1-9c79-c0e7e836bc13

**Comment by kdorosh** [comment]
/kick consul test flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=7a9331c3-a157-4101-9210-44c9aa6c7ec7

File: install/helm/gloo/templates/20-namespace-clusterrole-gateway.yaml [hunk 1]
```diff
@@ -200,13 +194,7 @@ rules:
...
   - list
   - watch
-  - update # needed for status updates
-- apiGroups:
-  - gateway.solo.io
-  resources:
-  - gateways
-  verbs:
-  - create # needed for creating the default gateways
+  - patch # needed for status updates
```
Comment (elcasteel) [inline install/helm/gloo/templates/20-namespace-clusterrole-gateway.yaml:197]: Is the stuff deleted from here general clean up or specific to the status change?

Comment (kdorosh) [reply install/helm/gloo/templates/20-namespace-clusterrole-gateway.yaml:197]: This change removes the permissions to create gateway CRs (general cleanup) which are no longer needed

Comment (kdorosh) [reply install/helm/gloo/templates/20-namespace-clusterrole-gateway.yaml:197]: Other changes in this file (all the create -> patch changes) are related to writing status using k8s patch

File: projects/gateway/pkg/syncer/translator_syncer.go [hunk 2]
```diff
@@ -396,15 +397,24 @@ func (s *statusSyncer) syncStatus(ctx context.Context) error {
...
+			// while tempting to write statuses in parallel to increase performance, we should actually consider recommending the user tunes k8s qps/burst:
+			// https://github.com/solo-io/gloo/blob/a083522af0a4ce22f4d2adf3a02470f782d5a865/projects/gloo/api/v1/settings.proto#L337-L350
 			if err := s.reporter.WriteReports(ctx, reports, currentStatuses); err != nil {
-				errs = multierror.Append(errs, err)
+				// add TEMPORARY warning to WriteReports that we should remove in Gloo Edge ~v1.16.0+.
+				// to get the status performance improvements, we need to make the assumption that the user has the latest CRDs installed.
+				// if a user forgets the error message is very confusing (invalid request during kubectl patch);
+				// this should help them understand what's going on in case they did not read the changelog.
+				errs = multierror.Append(errs, errors.Wrapf(err, "failed to write reports for %v;"+
+					"did you make sure your CRDs have been updated since v1.13.0-beta14? (i.e. `status` and `status.statuses` fields exist on your CR)", inputResource.GetMetadata().Ref().Key()))
```
Comment (elcasteel) [inline projects/gateway/pkg/syncer/translator_syncer.go:410]: If we backport this to 1.12 then we'll need to update the text here.

Comment (kdorosh) [reply projects/gateway/pkg/syncer/translator_syncer.go:410]: Good call out, we will

**Comment by elcasteel** [comment]
Gloo changes look good to me. I'm going to look at the the solo-kit side with fresh eyes tomorrow

**Review by elcasteel — APPROVED** [review]

**Pull request merged**